src/View/ViewLayerData.php

```diff
@@ -234,9 +234,11 @@ private function getMetaData(object $data, string $name): mixed
 
     private function callDataMethod(object $data, string $name, array $arguments, bool &$valueWasFetched = false): mixed
     {
+        $isModelData = $data instanceof ModelData;
         $hasDynamicMethods = method_exists($data, '__call');
         $hasMethod = ClassInfo::hasMethod($data, $name);
-        if ($hasMethod || $hasDynamicMethods) {
+        // While ModelData has dynamic methods, they should only be called if hasMethod returns true.
```

Comment:
```suggestion
        // ModelData has dynamic methods, though they should only be called if hasMethod returns true.
```

Could change PHP to one of these (up to you) or just leave as is

```php
$doCallMethod = $isModelData
  ? $hasMethod
  : $hasDynamicMethods || $hasMethod
;
  
$doCallMethod = match ($isModelData) {
  true: $hasMethod,
  false: $hasMethod || $hasDynamicMethods,
}

if ($isModelData) {
  $doCallMethod = $hasMethod;
} else {
  $doCallMethod = $hasDynamicMethods || $hasMethod
}
```

Reply:
Done - chose the `match` syntax.
I also added an additional line to the comment to hopefully help make it even clearer what's going on.